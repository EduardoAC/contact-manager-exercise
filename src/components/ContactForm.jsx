import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'

export const ContactForm = (props) => {
  const { onSubmit, handleSubmit, pristine, submitting } = props
  return (
    <form onSubmit={handleSubmit(onSubmit)} >
      <label>First name:</label>
      <div>
        <Field type="text" component="input" name="firstName" />
      </div>
      <label>Last name:</label>
      <div>
        <Field type="text" component="input" name="lastName" />
      </div>
      <label>Address:</label>
      <div>
        <Field type="text" component="input" name="address" />
      </div>
      <label>Phone number: </label>
      <div>
        <Field type="text" component="input" name="phoneNumber" />
      </div>
      <label>email: </label>
      <div>
        <Field type="text" component="input" name="email" />
      </div>
      <input type="submit" value="Submit" disabled={pristine || submitting} />
    </form>
  );
}

ContactForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool,
  pristine: PropTypes.bool,
}

export default reduxForm({
  form: 'contact'
})(ContactForm);
