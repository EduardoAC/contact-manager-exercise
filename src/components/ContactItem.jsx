import React from 'react';
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom';

const ContactItem = (props) => {
  const {
    id,
    firstName,
    lastName,
    phoneNumber,
    address,
    email,
  } = props;
  return (
    <tr className="contact__item">
      <td>{id}</td>
      <td>{firstName}</td>
      <td>{lastName}</td>
      <td>{address}</td>
      <td>{phoneNumber}</td>
      <td>{email}</td>
      <td>
        <Link to={`/${id}/edit`} style={{ marginRight: '5px' }}>Edit</Link>
        <Link to={`/${id}/delete`}>Delete</Link>
      </td>
    </tr>
  )
};

ContactItem.propTypes = {
  id: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default ContactItem;
