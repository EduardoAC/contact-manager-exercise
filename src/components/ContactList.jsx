import React from 'react';
import PropTypes from 'prop-types'

import ContactItem from './ContactItem';

const ContactList = (props) => {
  const { contacts } = props;
  const contactList = contacts.map(
    (contact) => (<ContactItem key={`contact-${contact.id}`} {...contact} />)
  );
  return (
    <div>
      <h1>Contact List</h1>
      <table>
        <tbody>
          {contactList}
        </tbody>
      </table>
    </div>
  );
};

ContactList.propTypes = {
  contacts: PropTypes.arrayOf(PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  })).isRequired,
};

export default ContactList;
