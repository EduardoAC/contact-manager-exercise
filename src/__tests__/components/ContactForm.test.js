import React from 'react';
import { shallow, mount } from 'enzyme';

import { ContactForm } from '../../components/ContactForm';

const componentProps = {
  onSubmit: jest.fn(),
  handleSubmit: jest.fn((cb, values) => cb(values)),
  submitting: false,
  pristine: false,
}
describe('<ContactForm />', () => {
  let wrapper
  it('Component renders correctly', () => {
    const wrapper = shallow(<ContactForm {...componentProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should submit the form', () => {
    const onSubmit = jest.fn();
    const specificProps = {...componentProps, onSubmit};
    const wrapper = shallow(<ContactForm {...specificProps} />);
    const submitButton = wrapper.find("input[type='submit']");
    submitButton.simulate('submit');
    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
