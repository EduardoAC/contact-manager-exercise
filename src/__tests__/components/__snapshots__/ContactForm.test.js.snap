// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    onSubmit={undefined}
>
    <label>
        First name:
    </label>
    <div>
        <Field
            component="input"
            name="firstName"
            type="text"
        />
    </div>
    <label>
        Last name:
    </label>
    <div>
        <Field
            component="input"
            name="lastName"
            type="text"
        />
    </div>
    <label>
        Address:
    </label>
    <div>
        <Field
            component="input"
            name="address"
            type="text"
        />
    </div>
    <label>
        Phone number: 
    </label>
    <div>
        <Field
            component="input"
            name="phoneNumber"
            type="text"
        />
    </div>
    <label>
        email: 
    </label>
    <div>
        <Field
            component="input"
            name="email"
            type="text"
        />
    </div>
    <input
        disabled={false}
        type="submit"
        value="Submit"
    />
</form>,
  "nodes": Array [
    <form
      onSubmit={undefined}
>
      <label>
            First name:
      </label>
      <div>
            <Field
                  component="input"
                  name="firstName"
                  type="text"
            />
      </div>
      <label>
            Last name:
      </label>
      <div>
            <Field
                  component="input"
                  name="lastName"
                  type="text"
            />
      </div>
      <label>
            Address:
      </label>
      <div>
            <Field
                  component="input"
                  name="address"
                  type="text"
            />
      </div>
      <label>
            Phone number: 
      </label>
      <div>
            <Field
                  component="input"
                  name="phoneNumber"
                  type="text"
            />
      </div>
      <label>
            email: 
      </label>
      <div>
            <Field
                  component="input"
                  name="email"
                  type="text"
            />
      </div>
      <input
            disabled={false}
            type="submit"
            value="Submit"
      />
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ContactForm
        handleSubmit={[Function]}
        onSubmit={[Function]}
        pristine={false}
        submitting={false}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleSubmit": [Function],
          "onSubmit": [Function],
          "pristine": false,
          "submitting": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          onSubmit={undefined}
>
          <label>
                    First name:
          </label>
          <div>
                    <Field
                              component="input"
                              name="firstName"
                              type="text"
                    />
          </div>
          <label>
                    Last name:
          </label>
          <div>
                    <Field
                              component="input"
                              name="lastName"
                              type="text"
                    />
          </div>
          <label>
                    Address:
          </label>
          <div>
                    <Field
                              component="input"
                              name="address"
                              type="text"
                    />
          </div>
          <label>
                    Phone number: 
          </label>
          <div>
                    <Field
                              component="input"
                              name="phoneNumber"
                              type="text"
                    />
          </div>
          <label>
                    email: 
          </label>
          <div>
                    <Field
                              component="input"
                              name="email"
                              type="text"
                    />
          </div>
          <input
                    disabled={false}
                    type="submit"
                    value="Submit"
          />
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          onSubmit={undefined}
>
          <label>
                    First name:
          </label>
          <div>
                    <Field
                              component="input"
                              name="firstName"
                              type="text"
                    />
          </div>
          <label>
                    Last name:
          </label>
          <div>
                    <Field
                              component="input"
                              name="lastName"
                              type="text"
                    />
          </div>
          <label>
                    Address:
          </label>
          <div>
                    <Field
                              component="input"
                              name="address"
                              type="text"
                    />
          </div>
          <label>
                    Phone number: 
          </label>
          <div>
                    <Field
                              component="input"
                              name="phoneNumber"
                              type="text"
                    />
          </div>
          <label>
                    email: 
          </label>
          <div>
                    <Field
                              component="input"
                              name="email"
                              type="text"
                    />
          </div>
          <input
                    disabled={false}
                    type="submit"
                    value="Submit"
          />
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ContactForm
    handleSubmit={[Function]}
    onSubmit={[Function]}
    pristine={false}
    submitting={false}
/>,
}
`;
